druid/src/widget/textbox.rs

```diff
@@ -219,6 +215,9 @@ impl TextBox {
 
 impl Widget<String> for TextBox {
     fn event(&mut self, ctx: &mut EventCtx, event: &Event, data: &mut String, env: &Env) {
+        // Guard against external changes in data
+        self.selection = self.selection.constrain_to(data);
```

Comment:
hmm, can we put this in `update`?

Reply:
That was the first thing I tried, but when you have a `TextBox` inside a `WidgetPod` inside a widget that modifies data after `event`, and it happens to modify it back to the data that it was before an event, `WidgetPod` would skip the `update` call as redundant and the internal state would get stuck out of sync.

Reply:
Interesting. good explanation, thanks.